src/clr-angular/utils/smart-popover/providers/smart-popover-events.service.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ *
+ */
+
+import { ElementRef, Injectable, Renderer2, Inject } from '@angular/core';
+import { ClrSmartPopoverToggleService } from './smart-popover-toggle.service';
+import { Observable, fromEvent, Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { DOCUMENT } from '@angular/common';
+
+// https://github.com/angular/angular/issues/20351#issuecomment-344009887
+/** @dynamic */
+@Injectable()
+export class ClrSmartPopoverEventsService {
+  public outsideClickClose = true;
+  public scrollToClose = true;
+  private documentClickListener: () => void;
+  public ignoredEvent: any;
+
+  constructor(
+    private renderer: Renderer2,
+    private smartOpenService: ClrSmartPopoverToggleService,
+    @Inject(DOCUMENT) private document: HTMLDocument
+  ) {
+    smartOpenService.openChange.subscribe(open => {
```

Comment:
Unhandled subscription, services also have a `ngOnDestroy` that hooks into the provider lifecycle.

Reply:
Added subscriptions array to manage these.